drivers/i2c/i2c_dw.c

```diff
@@ -461,17 +461,22 @@ static int i2c_dw_transfer(const struct device *dev,
 		return 0;
 	}
 
+	ret = k_mutex_lock(&dw->bus_mutex, K_FOREVER);
```

Comment:
May I ask why a semaphore is preferred here?

Reply:
Primarily this is for consistency at this point, Zephyr i2c/spi drivers currently use semaphores as a form of a bus lock. So my short answer today is to maintain consistency with the other set of drivers in the tree.

The longer answer is a mutex with PI/boosting could be useful in some scenarios but a mutex has some added complexity surrounding them along with added code size cost tied to that. Maybe this could be revisited, but then it should be revisited for all drivers in the tree.